Bottom: 7bcab7ed6c7de815036a99d397129e39ce2ca095
Top:    1036d028bf0532fbf7919e1e67974a580d086f39
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-04-28 01:33:36 +0100

main.c: Don't create Output.pmGC and Output.bgGC at init time

These are not used by any non draw.c code, and draw.c always stashes
the one created by main.c and creates its own.


---

diff --git a/src/draw.c b/src/draw.c
index c041195..0af8445 100644
--- a/src/draw.c
+++ b/src/draw.c
@@ -1727,8 +1727,6 @@ void
 hid_expose_callback (HID * hid, BoxType * region, void *item)
 {
   HID *old_gui = gui;
-  hidGC savebg = Output.bgGC;
-  hidGC savefg = Output.fgGC;
   hidGC savepm = Output.pmGC;
 
   gui = hid;
@@ -1752,7 +1750,5 @@ hid_expose_callback (HID * hid, BoxType * region, void *item)
   gui->destroy_gc (Output.bgGC);
   gui->destroy_gc (Output.pmGC);
   gui = old_gui;
-  Output.fgGC = savefg;
-  Output.bgGC = savebg;
   Output.pmGC = savepm;
 }
diff --git a/src/main.c b/src/main.c
index ad5ee94..e216f6d 100644
--- a/src/main.c
+++ b/src/main.c
@@ -977,9 +977,7 @@ main (int argc, char *argv[])
   if (show_copyright)
     copyright ();
 
-  Output.bgGC = gui->make_gc ();
   Output.fgGC = gui->make_gc ();
-  Output.pmGC = gui->make_gc ();
 
   settings_post_process ();
